src/Segment.java

```diff
@@ -0,0 +1,100 @@
+import java.lang.reflect.Array;
+import java.sql.Date;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+
+public class Segment extends Point {
+
+    private  String lat_destination , lng_destination , id_ride_destination, timestamp_destination;
+    private double U, distance, delta_time_hours;
+    private  int new_timestamp = Integer.parseInt(timestamp);
+    private  float new_id_ride = Float.parseFloat(id_ride);
+    private  double new_lng = Double.parseDouble(lng);
+    private double new_lat = Double.parseDouble(lat);
+
+    public Segment(String lat,
+                   String lng,
+                   String id_ride,
```

Comment:
Latitude and longitude are a point, right? So you can pass Point Origin and Point Destination instead of lat,LNG and lat2 and  lng2